awx/ui_next/src/components/Lookup/MultiCredentialsLookup.jsx

```diff
@@ -1,146 +1,167 @@
-  }
+import AnsibleSelect from '@components/AnsibleSelect';
+import { FieldTooltip } from '@components/FormField';
+import { CredentialChip } from '@components/Chip';
+import VerticalSeperator from '@components/VerticalSeparator';
+import { getQSConfig, parseQueryString } from '@util/qs';
+import Lookup from './Lookup';
+import OptionsList from './shared/OptionsList';
 
-  componentDidMount() {
-    this.loadCredentialTypes();
-  }
+const QS_CONFIG = getQSConfig('credentials', {
+  page: 1,
+  page_size: 5,
+  order_by: 'name',
+});
 
-  async loadCredentialTypes() {
-    const { onError } = this.props;
-    try {
-      const { data } = await CredentialTypesAPI.read();
-      const acceptableTypes = ['machine', 'cloud', 'net', 'ssh', 'vault'];
-      const credentialTypes = [];
-      data.results.forEach(cred => {
-        acceptableTypes.forEach(aT => {
-          if (aT === cred.kind) {
-            // This object has several repeated values as some of it's children
-            // require different field values.
-            cred = {
-              id: cred.id,
-              key: cred.id,
-              kind: cred.kind,
-              type: cred.namespace,
-              value: cred.name,
-              label: cred.name,
-              isDisabled: false,
-            };
-            credentialTypes.push(cred);
-          }
-        });
-      });
-      this.setState({ credentialTypes });
-    } catch (err) {
-      onError(err);
-    }
-  }
+async function loadCredentialTypes() {
+  const { data } = await CredentialTypesAPI.read();
+  const acceptableTypes = ['machine', 'cloud', 'net', 'ssh', 'vault'];
+  return data.results.filter(type => acceptableTypes.includes(type.kind));
+}
 
-  async loadCredentials(params) {
-    const { selectedCredentialType } = this.state;
-    params.credential_type = selectedCredentialType.id || 1;
-    return CredentialsAPI.read(params);
-  }
+async function loadCredentials(params, selectedCredentialTypeId) {
+  params.credential_type = selectedCredentialTypeId || 1;
+  const { data } = await CredentialsAPI.read(params);
+  return data;
+}
 
-  toggleCredentialSelection(newCredential) {
-    const { onChange, credentials: credentialsToUpdate } = this.props;
+function MultiCredentialsLookup(props) {
+  const { history, tooltip, value, onChange, onError, i18n } = props;
+  const [credentialTypes, setCredentialTypes] = useState([]);
+  const [selectedType, setSelectedType] = useState(null);
+  const [credentials, setCredentials] = useState([]);
+  const [credentialsCount, setCredentialsCount] = useState(0);
 
-    let newCredentialsList;
-    const isSelectedCredentialInState =
-      credentialsToUpdate.filter(cred => cred.id === newCredential.id).length >
-      0;
+  useEffect(() => {
+    (async () => {
+      try {
+        const types = await loadCredentialTypes();
+        setCredentialTypes(types);
+        setSelectedType(
+          types.find(type => type.name === 'Machine') || types[0]
```

Comment:
I think you're going to want to look for `type.kind === 'ssh'` rather than looking at the name.  I could be mistaken but my gut tells me the name might change when translated to other languages.  I'm not positive about that but I know the `kind` won't change.